Software/Development/SemesterProject/App/referenceHandler.c

```diff
@@ -0,0 +1,317 @@
+    }break;
+    }
+    return torqueReferenceBeforeLimit;
+}
+
+float CalculateTorqueRefFromAnalog(float analogRead)
+{
+    return analogRead/MAX_VALUE_ADC*MAX_TORQUE_REF_NM;
+}
+
+float calculateLimitedTorqueReference(float torqueReferenceBeforeLimit)
+{
+    if (referenceIncreased(torqueReferenceBeforeLimit)) return torqueReference + deltaTorque;
+    if (referenceDecreased(torqueReferenceBeforeLimit)) return torqueReference - deltaTorque;
+    else    return torqueReference;
+}
+
+int referenceIncreased(float torqueReferenceBeforeLimit)
+{
+    return torqueReferenceBeforeLimit > torqueReference + deltaTorque;
+}
+
+int referenceDecreased(float torqueReferenceBeforeLimit)
+{
+    return torqueReferenceBeforeLimit < torqueReference - deltaTorque;
+}
+
+float calculateSaturatedTorque(float torqueBeforeSaturation)
+{
+    if (torqueBeforeSaturation > MAX_TORQUE_REF_NM) return MAX_TORQUE_REF_NM;
+    else return torqueBeforeSaturation;
+}
+
+void calculateSpeedReference(void)
+{
+    int speedReferenceBeforeLimit = 0, speedReferenceBeforeSaturation = 0;
+    speedReferenceBeforeLimit = getSpeedReferenceValue();
+    speedReferenceBeforeSaturation = calculateLimitedSpeedReference(speedReferenceBeforeLimit);
+    speedReference = calculateSaturatedSpeed(speedReferenceBeforeSaturation);
+}
+int getSpeedReferenceValue(void)
+{
+    if (speedMustBeIncreased())     undampedSpeedReference += SPEED_STEP_FROM_BUTTON_RPM;
+    if (speedMustBeDecreased())     undampedSpeedReference -= SPEED_STEP_FROM_BUTTON_RPM;
+
+    return undampedSpeedReference;
+}
+
+int calculateLimitedSpeedReference(int speedReferenceBeforeLimit)
+{
+    if (speedRefIncreased(speedReferenceBeforeLimit)) return speedReferenceBeforeLimit + deltaSpeed;
+    if (speedRefDecreased(speedReferenceBeforeLimit)) return speedReferenceBeforeLimit - deltaSpeed;
+    else return speedReferenceBeforeLimit;
+}
+
+int speedRefIncreased(int speedReferenceBeforeLimit)
+{
+    return speedReferenceBeforeLimit > speedReference + deltaSpeed;
+}
+
+int speedRefDecreased(int speedReferenceBeforeLimit)
+{
+    return speedReferenceBeforeLimit < speedReference - deltaSpeed;
+}
+
+int speedMustBeIncreased(void)
+{
+    return speedRefIncreaseHasBeenPressed();
+}
+
+int speedMustBeDecreased(void)
+{
+    return speedRefDecreaseHasBeenPressed();
+    }
+
+int calculateSaturatedSpeed(int speedReferenceBeforeSaturation)
+{
+    if (speedReferenceBeforeSaturation > MAXIMUM_SPEED_REF_RPM) return MAXIMUM_SPEED_REF_RPM;
+    else return speedReferenceBeforeSaturation;
+}
+
+void restartSpeedReference(void)
+{
+    undampedSpeedReference = 0;
+}
+
+/*
+ * External software components interface
+ */
+
+float getTorqueReference(void)
+{
+    return torqueReference;
+}
+
+float getSpeedReference(void)
+{
+    return speedReference;
+}
+
+float getOpenLoopReference(void) //Provide a value between 0 and 100
+{
+    return torqueReference / MAX_TORQUE_REF_NM * 100;
+}
+
+int torqueControlIsEnabled(void)
```

Comment:
better name would be as a question?
isTorqueControlEnabled()?¿

Reply:
actually if you consider its use it, probably makes more sense as it is:

if(torqueControlIsEnabled())

Reply:
Ok, sounds valid